core/jobs/types/model_property.py

```diff
@@ -31,7 +38,14 @@ class ModelProperty:
 
     __slots__ = ('_model_kind', '_property_name')
 
-    def __init__(self, model_class, property_obj):
+    # Here, argument property_obj accepts a property of a model and the
+    # property of a model can be a object of str, int, datetime and other
+    # objects too. So, Any type has to be used for property_obj argument.
+    def __init__(
+        self,
+        model_class: Type[base_models.BaseModel],
+        property_obj: Any
```

Comment:
This argument can receive the property of a modal and that property can be of any kind of object like: string_object, Int object, and other types of objects too. That's why I used Any type here.

Reply:
(please correct me if i'm wrong)
By looking at the code, it is clear that `property_obj` can accept all the properties of type `datastore_services.Property` but it can accept only 1 python property which is `id`. So I think the type of `property_obj` should be 
`Union[datastore_services.Property, Callable[[base_models.BaseModel], str]]`. Here, i used `Callable[[base_models.BaseModel], str]` for python `id` property because, the property (python property) class is used in code as a decorator and the type of value returned by property (id) is `str`. So that's why `Callable[[base_models.BaseModel], str]` is used. 
Thanks!!